manager/controllers/app/blueprint_mgr.go

```diff
@@ -27,10 +26,16 @@ func containsTemplate(templateList []app.ComponentTemplate, moduleName string) b
 
 // GetBlueprintSignature returns the signature (name and namespace) of the blueprint owned by the given M4DApplication instance
 func (r *M4DApplicationReconciler) GetBlueprintSignature(applicationContext *app.M4DApplication) *app.Blueprint {
+	//sanity - finalizer should exist, since it represents the blueprint namespace name
```

Comment:
The use of finalizer name as the namespace indicator seems odd here isn't it? I always thought that the finalizer name should typically represent the controller itself, so the controller would have a chance to run all of its cleaning logic. Reusing it for storing the namespace seems a bit like an abuse.

The code related to finalizer is already not so clean so I suggest to consider the following:
1. Revert the changes in this PR and use a Status field instead to indicate the generated namespace name
2. Keep the code around finalizers independent of the cleanup logic and place the cleanup logic in a separate function (like `deleteExternalResources` in https://book.kubebuilder.io/reference/using-finalizers.html) that will handle all cleaning, including the namespace if indicated in the status.

Reply:
Done.